### PR TITLE
Support secrets from v2 kv store

### PR DESCRIPTION
**Pull request by luke-richardson**
The v2 kv secrets engine returns the secret data in a different format (https://www.vaultproject.io/api/secret/kv/kv-v2.html#sample-response-1) than v1, this means that vault-sidekick writes a string representation of a map with "data" and "metadata" keys to file, rather than the secret data.

This change introduces a check to identify if the secrets engine is a v1 or v2 response.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,6 +74,7 @@ spec:
       - -cn=pki:project1/certs/example.com:common_name=commons.example.com,revoke=true,update=2h
       - -cn=secret:secret/db/prod/username:file=.credentials
       - -cn=secret:secret/db/prod/password:retries=true
+      - -cn=secret:scecret/data/db/dev/username:file=.kv2credentials
```
Comment (gambol99) [inline README.md]: s/scecret/secret/

File: vault.go [hunk 2]
```diff
@@ -414,6 +414,12 @@ func (r VaultService) get(rn *watchedResource) error {
 				secret, err = r.client.Logical().Read(rn.resource.path)
 			}
 		}
+		// if there is a top-level metadata key this is from a v2 kv store
```
Comment (gambol99) [inline vault.go]: adding a link to docs for reference https://www.vaultproject.io/api/secret/kv/kv-v2.html

File: README.md (hunk 1 above)
Comment (luke-richardson) [reply README.md]: corrected!

**Comment by gambol99** [comment]
Sorry for the massive delay @luke-richardson .. thank you kindly .. LGTM

**Pull request merged**